.github/workflows/build-docs.yml

```diff
@@ -36,42 +35,73 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  build-docs:
+  Check-for-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      trigger-check-outcome: ${{ steps.trigger_check.outcome }}
+      docs-check-outcome: ${{ steps.docs_check.outcome }}
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name != 'pull_request_target'
+
+      - name: Checkout pull/${{ github.event.number }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+        if: github.event_name == 'pull_request_target'
+
+      - name: Check for trigger by push event, manual dispatch, build-docs label on a PR
+        id: trigger_check
+        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'pull_request_target' && contains(github.event.pull_request.labels.*.name, 'build-docs')
+        run: |
+          echo "Building docs as a test."
+          exit 0
+        continue-on-error: true
 
-    if: github.event_name == 'push' ||
-        github.event_name == 'workflow_dispatch' ||
-        (github.event_name == 'pull_request_target' &&
-        contains(github.event.pull_request.labels.*.name, 'build-docs'))
+      - name: Check for changes in documentation
+        run: |
+          if git diff origin/master..."$(git rev-parse --abbrev-ref HEAD)" --name-only | cat | grep '^docs/' | grep -q .; then
+              num_files=$(git diff --name-only origin/master...HEAD | grep '^docs/' | wc -l)
+              echo "Changes found in documentation files: $num_files"
+              exit 0
+          else
+              echo "No changes found in documentation files - will stop running the pipeline."
+              exit 1
+          fi
+        id: docs_check
+        if: steps.trigger_check.outcome != 'success'
+        continue-on-error: true
 
+  build-docs:
     runs-on: ubuntu-latest
+    needs: Check-for-changes
+    if: needs.Check-for-changes.outputs.trigger-check-outcome == 'success' || needs.Check-for-changes.outputs.docs-check-outcome == 'success'
```

Comment:
```suggestion
    if: needs.check-for-changes.outputs.trigger-check-outcome == 'success' || needs.check-for-changes.outputs.docs-check-outcome == 'success'
```

Reply:
I have renamed the job using lower case.